packages/code-du-travail-frontend/src/outils/utils/extractDates.ts

```diff
@@ -19,7 +19,7 @@ export const convertPeriodToHumanDate = (
   if (!unit) return null;
   let extra = getExtra(input);
   if (unit === Unit.DAY && extra !== Extra.FROM_TO) value--;
-  date = convertDate(from, value, unit, !extra);
+  date = convertDate(from, value, unit, extra === Extra.FROM_TO);
```

Comment:
c'est pas plutot ça ?

```suggestion
  date = convertDate(from, value, unit, extra !== Extra.FROM_TO);
```

Reply:
Non, c'est bien ça, on applique la logique si l'extra est en `de date à date`, dans les autres cas ce n'est pas le cas. Cette logique métier, je l'ai fait à travers les paramètres de la fonction, comme tu peux le voir ci-dessous :

```ts
export const convertDate = (
  date: Date,
  value: number,
  unit: Unit,
  isDateADate = false
): Date => {
```


Est-ce que ça  serait pas mieux de passer l'extra, et de mettre la logique côté fonciton. Honnêtement, je pense que c'est mieux car le code n'est pas clair quand on le lit. Pour bien comprendre, il faut lire les paramètres de la fonction ce qui n'est pas optimal.

J'applique donc cela si ça te va

Reply:
Bon du coup, j'ai remis comme avant car le métier a changé d'avis